pkg/conduit/entrypoint.go

```diff
@@ -20,22 +20,32 @@ import (
 	"fmt"
 	"os"
 	"os/signal"
-	"strings"
 
 	"github.com/conduitio/conduit/pkg/foundation/cerrors"
 	"github.com/peterbourgon/ff/v3"
 	"github.com/peterbourgon/ff/v3/ffyaml"
 )
 
-var DeprecatedFlags = map[string]bool{
-	"pipelines.exit-on-error": true,
-}
-
 const (
 	exitCodeErr       = 1
 	exitCodeInterrupt = 2
+
+	FlagDevCpuProfile   = "dev.cpuprofile"
+	FlagDevMemProfile   = "dev.memprofile"
+	FlagDevBlockProfile = "dev.blockprofile"
+
+	// Deprecated: Use `pipelines.error-recovery.exit-on-degraded` instead.
+	FlagPipelinesExitOnError = "pipelines.exit-on-error"
 )
 
+// HiddenFlags is a map of flags that should not be shown in the help output.
+var HiddenFlags = map[string]bool{
+	FlagPipelinesExitOnError: true,
+	FlagDevCpuProfile:        true,
+	FlagDevMemProfile:        true,
+	FlagDevBlockProfile:      true,
+}
```

Comment:
Just to share a thought (I don't think we need to change anything now): this looks good to me, but what I liked about the previous version (where we checked for the `dev` prefix) is that it was a rule and not an enumeration. That way, if we ever add a new dev flag, it will automatically be excluded.

Reply:
You got a point. Will refactor.